win32/tests/file_win32_ut/file_win32_ut.c

```diff
@@ -0,0 +1,952 @@
+    STRICT_EXPECTED_CALL(mock_SetFileCompletionNotificationModes(handle, FILE_SKIP_COMPLETION_PORT_ON_SUCCESS))
+        .SetReturn(TRUE);
+    STRICT_EXPECTED_CALL(mock_InitializeThreadpoolEnvironment(IGNORED_ARG))
+        .CaptureArgumentValue_pcbe(&captured_pcbe);
+    STRICT_EXPECTED_CALL(execution_engine_win32_get_threadpool(execution_engine))
+        .SetReturn(ptp_pool);
+    STRICT_EXPECTED_CALL(mock_SetThreadpoolCallbackPool(IGNORED_ARG, ptp_pool))
+        .ValidateArgumentValue_pcbe(&captured_pcbe);
+    STRICT_EXPECTED_CALL(mock_CreateThreadpoolCleanupGroup())
+        .SetReturn(ptp_cleanup_group);
+    STRICT_EXPECTED_CALL(mock_SetThreadpoolCallbackCleanupGroup(IGNORED_ARG, ptp_cleanup_group, IGNORED_ARG));
+    STRICT_EXPECTED_CALL(mock_CreateThreadpoolIo(handle, IGNORED_ARG, NULL, IGNORED_ARG))
+        .CaptureArgumentValue_pcbe(&captured_pcbe)
+        .SetReturn(ptp_io);
+
+    umock_c_negative_tests_snapshot();
+    for (int i = 0; i < umock_c_negative_tests_call_count(); ++i)
+    {
+        if (umock_c_negative_tests_can_call_fail(i))
+        {
+            umock_c_negative_tests_reset();
+            umock_c_negative_tests_fail_call(i);
+
+            ///act
+            FILE_HANDLE file_handle = file_create(execution_engine, filename, NULL, NULL);
+
+            ///assert
+            ASSERT_IS_NULL(file_handle);
+        }
+    }
+}
+
+/*Tests_SRS_FILE_43_005: [ If handle is NULL, file_destroy shall return. ]*/
+/*Tests_SRS_FILE_WIN32_43_049: [ If handle is NULL, file_destroy shall return. ]*/
+TEST_FUNCTION(file_destroy_called_with_null_handle)
+{
+    ///arrange
+
+    ///act
+    file_destroy(NULL);
+
+    ///assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+}
+
+/*Tests_SRS_FILE_WIN32_43_011: [ file_destroy shall wait for all I / O to complete by calling WaitForThreadpoolIoCallbacks.]*/
+/*Tests_SRS_FILE_WIN32_43_012: [ file_destroy shall close the cleanup group by calling CloseThreadpoolCleanupGroup.]*/
+/*Tests_SRS_FILE_WIN32_43_013: [ file_destroy shall destroy the environment by calling DestroyThreadpoolEnvironment.]*/
+/*Tests_SRS_FILE_WIN32_43_016: [ file_destroy shall call CloseHandle on handle->h_file]*/
+/*Tests_SRS_FILE_WIN32_43_015: [ file_destroy shall close the threadpool IO by calling CloseThreadPoolIo.]*/
+/*Tests_SRS_FILE_WIN32_43_042: [ file_destroy shall free the handle.]*/
+TEST_FUNCTION(file_destroy_succeeds)
+{
+
+    PTP_WIN32_IO_CALLBACK captured_callback;
+    FILE_HANDLE file_handle = get_file_handle("file_destroy_succeeds.txt", &captured_callback);
+    PTP_IO captured_ptp_io;
+
+    STRICT_EXPECTED_CALL(mock_WaitForThreadpoolIoCallbacks(IGNORED_ARG, FALSE))
+        .CaptureArgumentValue_pio(&captured_ptp_io);
+    STRICT_EXPECTED_CALL(mock_CloseThreadpoolCleanupGroup(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(mock_DestroyThreadpoolEnvironment(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(mock_CloseHandle(IGNORED_ARG));

```

Comment:
>IGNORED_ARG [](start = 42, length = 11)

These should all be those fixed global values I mentioned in the comment about the `get_file_handle`